File: 1 - General/Vishal/Vishal S/Easy/Find Pivot Index_Find the Middle Index in Array/Solution.go
func pivotIndex(nums []int) int {
    nums = append([]int{0},nums...)
    for i:=1;i<len(nums);i++ {
        nums[i] += nums[i-1]
    }
    l := len(nums)-1
    for i:=1;i<len(nums);i++ {
        if nums[l]-nums[i]==nums[i-1] {
            return i-1
        }
    }
    return -1
}

File: 1 - General/Vishal/Vishal S/Easy/Find Pivot Index_Find the Middle Index in Array/Solution.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        l = len(nums)
        for i in range(1,l):
            nums[i] += nums[i-1]
        
        for i in range(l):
            prev = nums[i-1] if i-1 >=0 else 0
            nxt = (nums[-1]-nums[i]) if i != (l-1) else 0
            if  prev == nxt:
                return i
        return -1